src/resolvers/directives/requireOneOfRoles.ts

```diff
@@ -1,33 +1,49 @@
 import { DirectiveResolverMap } from '../../typings/resolverMap';
 import { oneLine } from 'common-tags';
 import { ApiError } from '../../helpers/apiError';
+import { resolvers as requireAuthResolvers } from './requireAuth';
+
+const { requireAuth } = requireAuthResolvers;
 
 export const resolvers: Pick<DirectiveResolverMap, 'requireOneOfRoles'> = {
-  async requireOneOfRoles(next, _source, { roles }, { viewer }) {
-    if (!viewer) {
-      throw new ApiError('NOT_AUTHENTICATED');
-    }
+  requireOneOfRoles: async (next, source, { roles }, context, info) => {
+    const checkForMatchingRoles = async () => {
+      const { viewer } = context;
+      if (!viewer) {
+        throw new ApiError('NOT_AUTHENTICATED');
+      }
+
+      if (viewer.role === null) {
+        throw new ApiError('NOT_AUTHORIZED');
+      }
 
-    if (viewer.role === null) {
-      throw new ApiError('NOT_AUTHORIZED');
-    }
+      if (viewer.role && roles.includes(viewer.role)) {
+        return next();
+      }
 
-    if (viewer.role && roles.includes(viewer.role)) {
-      return next();
-    }
+      let errorMessage: string;
+      if (roles.length === 1) {
+        errorMessage = `Only users with an "${
```

Comment:
What are the possible `roles` values? Is the value guaranteed to start with a vowel to justify the "an"?

Reply:
I don't know what I was thinking :thinking:, this needs to be rephrased to avoid having to deal with a/an, and by the way it's not as simple as "use an if the word starts with a vowel". I just recently learned that. Check out this SO thread: https://stackoverflow.com/posts/15366802/revisions